controllers/discovery/initDiscoveryUserData.js

```diff
@@ -12,7 +12,7 @@ const {MINIMUM_KARMA_SCORE, MAX_EXCLUDED_USERS} = require('../../helpers/constan
  */
 const InitDiscoveryUserData = async (req, res) => {
   let {limit = 50, page = 0, allow_anon_dm} = req.query;
-  let {excluded_users = []} = req.body;
+  let {excluded_user_ids = [], excluded_user_names = []} = req.body;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The change from `excluded_users` to `excluded_user_ids` and `excluded_user_names` is a good one as it provides more flexibility in excluding users. However, there's no validation of the input data. It would be better to validate that `excluded_user_ids` are indeed numbers and `excluded_user_names` are strings to prevent potential SQL injection attacks or other issues.
```diff
- let {excluded_user_ids = [], excluded_user_names = []} = req.body;
+ let {excluded_user_ids = [], excluded_user_names = []} = req.body;
+ // Validate input data
+ if (!Array.isArray(excluded_user_ids) || !excluded_user_ids.every(Number.isInteger)) {
+   return res.status(400).json({ error: 'Invalid excluded_user_ids' });
+ }
+ if (!Array.isArray(excluded_user_names) || !excluded_user_names.every(name => typeof name === 'string')) {
+   return res.status(400).json({ error: 'Invalid excluded_user_names' });
+ }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->